### PR TITLE
Update supported Rubies

### LINKED ISSUE #94: End support for unused Ruby / Rails versions
We currently support usage of this gem in the context of both Ruby 1.9.3 and Rails 3.2 but our applications and services have since moved on from here. We should end support for these versions.

### PR DESCRIPTION
**Pull request by benlovell**
This PR is an RFC for removing Ruby `1.9.3`. I've also added preliminary support for Ruby `2.3.0` since we now include it as part of our provisioning stack and it's the latest released version of Ruby.

I've bumped the default project Ruby to `2.2.4` but I guess we could move this further along to `2.3.0` depending on consensus. `¯\_(ツ)_/¯`

Closes #94

### CODE REVIEW

**Comment by tijmenb** [comment]
Are we sure about dropping rails 3.2? `panopticon`, `publisher` all seem to run 3.2.22:

https://docs.google.com/spreadsheets/d/1FJmr39c9eXgpA-qHUU6GAbbJrnenc0P7JcyY2NB9PgU/edit#gid=1480786499

**Comment by benlovell** [comment]
@tijmenb Sure, I did allude to that in 2a85fe8, which will be precluded by those apps. We can either leave this hanging around until such a time or I'll happily clobber 2a85fe8 if we want to get this merged as-is.

**Comment by bradwright** [comment]
> Edit: Those apps seem to be the ones blocked by govuk_content_models which now (almost) supports Rails >= 4.0

They're blocked by that Gem bump *and* by anyone scheduling further work to get that upgraded. Given how critical SSO is to our security I'd prefer we didn't drop support for any apps currently in production.

**Comment by benlovell** [comment]
@bradwright Makes sense to drop 2a85fe8 for the time being. I'll do that.

**Comment by benlovell** [comment]
@tijmenb @bradwright Dropped 2a85fe8.

File: jenkins.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ rm -f gemfiles/*.gemfile.lock
 # Exclude /tmp from git clean as it only contains the signonotron checkout
 git clean -fdxe /tmp
 
-for ruby_version in 1.9.3 2.1 2.2; do
+for ruby_version in 2.1 2.2 2.3; do
   for gemfile in rails_3.2 rails_4.0 rails_4.1 rails_4.2; do
```
Comment (jamiecobbett) [inline jenkins.sh]: According to [the spreadsheet](https://docs.google.com/spreadsheets/d/1FJmr39c9eXgpA-qHUU6GAbbJrnenc0P7JcyY2NB9PgU/edit#gid=1480786499), we can drop Rails 4.0.

Comment (benlovell) [inline jenkins.sh]: Sweet. I'll do that now.

Comment (benlovell) [inline jenkins.sh]: @jamiecobbett f78af72

**Pull request merged**